spec/library_publish.spec.js

```diff
@@ -6,142 +6,47 @@ import {expect} from 'chai';
-
-		describe('that is a v2 library', () => {
-			const libraryName = 'test-library-publish';
-			let resourcesDirectory;
-			before(() => {
-				resourcesDirectory = require('particle-library-manager').resourcesDir;
-			});
-
-			it('can delete the existing library', () => {
-				return libraryDelete(libraryName);
-			});
-
-			it('can publish the library', () => {
-				const version = '0.0.1';
-				copySync(path.join(resourcesDirectory(), 'libraries', 'publish', 'valid', version), context.projectDir);
-				const module = require('../lib/library_publish');
-				const publish = expectPublishLibrary(module.notifyLibraryPublished({name:libraryName, metadata: {version}}));
-				return publish;
-			});
-
-			it('cannot republish the same version of the library', () => {
-				const version = '0.0.1';
-				copySync(path.join(resourcesDirectory(), 'libraries', 'publish', 'valid', version), context.projectDir);
-				return expect(runPublish(() => {})).to.eventually.be.rejectedWith('This version already exists. Version must be greater than 0.0.1');
-			});
-
-			it('can publish the library with a new version', () => {
-				const version = '0.0.2';
-				copySync(path.join(resourcesDirectory(), 'libraries', 'publish', 'valid', version), context.projectDir);
-				const module = require('../lib/library_publish');
-				const publish = expectPublishLibrary(module.notifyLibraryPublished({name:libraryName, metadata: {version}}));
-				return publish;
-			});
-
-			it('can delete the existing library as cleanup', () => {
-				return libraryDelete(libraryName);
-			});
-
-		});
-	});
+    function expectValidationErrorNotification(msg) {
+        const module = require('../lib/library_contribute');
+        const expected = module.notifyValidationError(msg);
+        expectNotification(expected);
+    }
+
+    function runPublish(then) {
+        return runCommand('publish', then);
+    }
+
+    describe('and library publish is run in a directory', () => {
+        describe('that is a v2 library', () => {
+            const libraryName = 'test-library-publish';
+            let resourcesDirectory;
+            before(() => {
+                resourcesDirectory = require('particle-library-manager').resourcesDir;
+            });
+
+            it('can delete the existing library', () => {
+                return libraryDelete(libraryName);
+            });
+
+            it('can publish the library', () => {
+                const version = '0.0.1';
+                copySync(path.join(resourcesDirectory(), 'libraries', 'contribute', 'valid', version), context.projectDir);
+                const module = require('../lib/library_publish');
+                return runCommand('contribute', () => {
+                    console.log('CONTIRBUTE');
+                    return runPublish(() => {
+                        console.log('PUBLISH');
+                        return expectNotification(module.notifyLibraryPublished({name:libraryName+'z', metadata: {version}}));
```

Comment:
what's the +'z' here? looks like something i might have done to get a failing test?